helix-core/src/main/java/org/apache/helix/cloud/event/helix/AbstractCloudEventCallbackImpl.java

```diff
@@ -0,0 +1,46 @@
+package org.apache.helix.cloud.event.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.HelixManager;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+public abstract class AbstractCloudEventCallbackImpl {
```

Comment:
Sorry, I don't quite get it. This class is abstract so users can't directly instantiate it. And all method implementation are dummy.  Could you clarify why `NotImplementedException` is an acceptable minimum implementation?
I'm thinking of using interface with default methods, it may achieve the same with more interface flexibility.

Reply:
Sorry for the confusion here. @xyuanlu Is currently working on the implementation of these callbacks.